### PR TITLE
Issue/994 redis campaign get single cache

### LINKED ISSUE #994: Cache single campaigns requests
_BUG_
-------------------
#### Current Behavior
- Single campaign responses are not cached. 

#### Desired Behavior
- The campaign response should be cached for at least 1 hour by default.

#### Steps to Replicate
- Send an small size broadcast (~5k) to Conversations API at full capacity (19 1X-std Dynos). It overwhelms Campaigns API at 6 1X-std Dynos.

#### Why This Matters
When the Campaigns API crashes, it errors out on all import requests during a broadcast until it's restarted and scaled significantly.

### PR DESCRIPTION
**Pull request by rapala61**
#### What's this PR do?
It uses Redis to centralize campaign caching when Campaigns API uses the `getCampaignById` method in the `lib/phoenix.js` module.

#### How should this be reviewed?
- It will be extensibly tested in Staging before any merging can occur.
- 👀 
- Checkout branch locally.
    - Follow direction of the README to get Redis up and running.
    - GET a few campaigns through `/v1/campaigns/<CAMPAIGN-ID>` the first time it should log: `Campaigns cache miss:<CAMPAIGN-ID>`, subsequent tries should log `Campaigns cache hit:<CAMPAIGN-ID>`.

#### Any background context you want to provide?
This is the MVP integration with Redis.

#### Relevant tickets
Fixes #994

#### Checklist
- [x] Documentation added for new features/changed endpoints.
- [ ] Tested on staging.

### CODE REVIEW

**Comment by rapala61** [comment]
What a great time for Slack to be down. I have tested these changes against staging. Sent a broadcast to 34698  recipients at about ~60/rps while 1241 users responded at ~5/rps. It is nothing short of amazing the amount of resources we are saving by caching single campaigns with Redis. I went from using 19 Dynos to 6 in Convo API, and 7 from 12 in Campaigns API. Retaining similar performance as previous tests.

**Comment by aaronschachter** [comment]
That sounds awesome @rapala61  - great work! I'll be able to review / test a little later tonight.

File: README.md [hunk 1]
```diff
@@ -5,15 +5,20 @@ Gambit is the internal DoSomething.org API used to send/receive messages to and
 
 ### Getting Started
 
-Install Node, MongoDB, and the Heroku toolbelt.
+Install Node, MongoDB, redis, and the Heroku toolbelt.
 
 Next, fork and clone this repository. To run Gambit locally:
-* `sudo mongod`
+* Mongo should be runing.
```
Comment (aaronschachter) [inline README.md]: The grammar police in me can't resist: Typo.

File: test/lib/phoenix.test.js [hunk 2]
```diff
@@ -31,32 +31,36 @@ test.afterEach(() => {
...
+test('phoenix should respond to getCampaignById', () => {
+  phoenix.should.respondTo('getCampaignById');
 });
 
 test('phoenix should respond to isClosedCampaign', () => {
   phoenix.should.respondTo('isClosedCampaign');
 });
 
-test('phoenix.fetchCampaign should call parsePhoenixCampaign on successt', async () => {
+test('phoenix.fetchCampaignById should call parsePhoenixCampaign on successt', async () => {
```
Comment (aaronschachter) [inline test/lib/phoenix.test.js]: typo 🚓

**Pull request merged**